index-scheduler/src/utils.rs

```diff
@@ -2,24 +2,18 @@
 
 use meilisearch_types::heed::{types::DecodeIgnore, RoTxn, RwTxn};
 use meilisearch_types::milli::BEU32;
-use roaring::RoaringBitmap;
+use roaring::{MultiOps, RoaringBitmap};
 
 use crate::{Error, IndexScheduler, Result, Task, TaskId};
 use meilisearch_types::tasks::{Kind, Status};
 
 impl IndexScheduler {
     pub(crate) fn all_task_ids(&self, rtxn: &RoTxn) -> Result<RoaringBitmap> {
-        let mut all_tasks = RoaringBitmap::new();
-        for status in [
-            Status::Enqueued,
-            Status::Processing,
-            Status::Succeeded,
-            Status::Failed,
-        ] {
-            all_tasks |= self.get_status(&rtxn, status)?;
-        }
-        Ok(all_tasks)
+        enum_iterator::all()
+            .map(|s| self.get_status(&rtxn, s))
+            .r#union()
```

Comment:
```suggestion
            .union()
```

We don't need anything specific to handle the handle the `union` word as a method?

Reply:
But it breaks my editor syntax highlighting 😢 Ok... I'll remove that...